src/apps/competitions/tests/test_submissions.py

```diff
@@ -80,3 +80,38 @@ def test_submission_not_created_if_max_reached(self):
                 assert False, "Submission should not have been created"
             except Submission.DoesNotExist:
                 pass
+
+
+class SubmissionManagerTests(TestCase):
+    def setUp(self):
+        self.user = UserFactory(username='test')
+        self.competition = CompetitionFactory(created_by=self.user)
+        self.phase = PhaseFactory(competition=self.competition)
+
+    def make_submission(self, **kwargs):
+        kwargs.setdefault('owner', self.user)
+        kwargs.setdefault('phase', self.phase)
+        return SubmissionFactory(**kwargs)
+
+    def test_re_run_submission_creates_new_submission_with_same_data_owner_and_phase(self):
+        sub = self.make_submission()
+        assert Submission.objects.all().count() == 1
+        sub.re_run()
+        assert Submission.objects.all().count() == 2
+        subs = Submission.objects.all()
+        assert subs[0].owner == subs[1].owner
+        assert subs[0].phase == subs[1].phase
+        assert subs[0].data == subs[1].data
+
+    # TODO: Figure out why this test causes CircleCI to freeze
+    # def test_cancel_submission_sets_status(self):
+    #     sub = self.make_submission()
+    #     assert sub.cancel(), 'Cancel returned False, meaning the submission could not be cancelled when it should'
+    #     assert sub.status == 'Cancelled'
+
```

Comment:
@ckcollab Any idea why this one test causes CircleCI to freeze up? if it is uncommented, circle can't get through 'not e2e' tests, but as soon as I comment it out it all runs just fine?

Reply:
no idea off the top of my head :(

Reply:
I guess leaving this commented out and hoping that in the review something may become obvious as to why this tests breaks things and we can fix & uncomment.

Reply:
Can make an issue for this?

Reply:
- [x] issue created